packages/react-jss/src/createUseStyles.js

```diff
@@ -52,16 +52,20 @@ const createUseStyles = <Theme: {}>(styles: Styles<Theme>, options?: HookOptions
         sheetOptions
       })
 
-      if (context.registry && sheet) {
-        context.registry.add(sheet)
+      let dynamicRules
+      let classes
+      if (sheet) {
+        if (context.registry) {
+          context.registry.add(sheet)
+        }
+        dynamicRules = addDynamicRules(sheet, data)
+        classes = getSheetClasses(sheet, dynamicRules)
       }
 
-      const dynamicRules = sheet && addDynamicRules(sheet, data)
-
       return {
         sheet,
         dynamicRules,
-        classes: sheet ? getSheetClasses(sheet, dynamicRules) : {}
+        classes: classes || {}
```

Comment:
Classes should always be an object, so the user can safely access any props when `disableStylesGeneration` is enabled